packages/flutter_tools/lib/src/base/logger.dart

```diff
@@ -63,3 +49,72 @@ class BufferLogger implements Logger {
   void printStatus(String message) => _status.writeln(message);
   void printTrace(String message) => _trace.writeln(message);
 }
+
+class VerboseLogger implements Logger {
+  _LogMessage lastMessage;
+
+  bool get isVerbose => true;
+
+  void printError(String message, [StackTrace stackTrace]) {
+    _emit();
+    lastMessage = new _LogMessage(_LogType.error, message, stackTrace);
+  }
+
+  void printStatus(String message) {
+    _emit();
+    lastMessage = new _LogMessage(_LogType.status, message);
+  }
+
+  void printTrace(String message) {
+    _emit();
+    lastMessage = new _LogMessage(_LogType.trace, message);
+  }
+
+  void flush() => _emit();
+
+  void _emit() {
+    lastMessage?.emit();
+    lastMessage = null;
+  }
+}
+
+enum _LogType {
+  error,
+  status,
+  trace
+}
+
+class _LogMessage {
+  _LogMessage(this.type, this.message, [this.stackTrace]) {
+    stopwatch.start();
+  }
+
+  static const String _bold = '\u001B[1m';
```

Comment:
if the terminal is dumb (e.g. emacs compilation buffer), this will make the output ugly. you should check if TERM is "dumb" and not use this if so. There's code somewhere in the tools that does this already when deciding if to pass --color to some command IIRC.